Testable process termination

nfnt opened the pull request:
Moves process handling of the executor into a separate function. This
allows to have tests for process termination.

orlandohohmeier commented on crates/worker/src/driver/process.rs:
```diff
@@ -0,0 +1,147 @@
+        sleep(Duration::from_secs(1)).await;
+
+        token.cancel();
+        process.await.unwrap();
```
Wonder whether we should wrap this in a timeout as this shouldn't hang

nfnt committed:
refactor: testable process termination

Moves process handling of the executor into a separate function. This
allows to have tests for process termination.
